doc/tutorials/acrn-secure-boot-with-efi-stub.rst

```diff
@@ -0,0 +1,140 @@
+- ``acrn.efi`` unpacks ACRN Hypervisor image and VM Kernels from a stitched container blob
+- ``acrn.efi`` loads ACRN Hypervisor to memory
+- ``acrn.efi`` prepares MBI to store Service VM & pre-launched RTVM Kernel info
+- ``acrn.efi`` hands over control to ACRN Hypervisor with MBI
+- ACRN Hypervisor boots Service VM and pre-launched RTVM in parallel
+
+As the container blob format, ``ACRN EFI-Stub`` uses the `Slim Bootloader Container
+Boot Image <https://slimbootloader.github.io/how-tos/create-container-boot-image.html>`_.
+
+Verified Configurations
+***********************
+- ACRN Hypervisor Release Version 2.5
+- hybrid_rt scenario
+- TGL-U RVP board
+- CONFIG_MULTIBOOT2=y (as default)
+- CONFIG_RELOC=y (as default)
+
+Building
+********
+
+Build Dependencies
+==================
+
+- Build Tools and Dependencies described in the :ref:`getting-started-building` guide
+- gnu-efi package
+- Service VM Kernel bzImage
+- pre-launched RTVM Kernel bzImage
+- `Slim Bootloader Container Tool <https://slimbootloader.github.io/how-tos/create-container-boot-image.html>`_
+
+The Slim Bootloader Tools can be downloaded from its `GitHub project <https://github.com/slimbootloader/slimbootloader>`_.
+The verified version is the commit `9f146af <https://github.com/slimbootloader/slimbootloader/tree/9f146af>`_.
+You may use the `meta-acrn Yocto Project integration layer <https://github.com/intel/meta-acrn>`_ to build Service VM Kernel and pre-launched one.
+
+Build EFI-Stub for TGL-U RVP hybrid_rt
+======================================
+
+.. code-block:: none
+
+   $ TOPDIR=`pwd`
+   $ cd acrn-hypervisor
+   $ make BOARD=tgl-rvp SCENARIO=hybrid_rt hypervisor
+   $ make BOARD=tgl-rvp SCENARIO=hybrid_rt -C misc/efi-stub/ \
+     HV_OBJDIR=`pwd`/build/hypervisor/ EFI_OBJDIR=`pwd`/build/hypervisor/misc/efi-stub `pwd`/build/hypervisor/misc/efi-stub/boot.efi
+
+Create Container
+================
+
+.. code-block:: none
+
+   $ mkdir -p $TOPDIR/acrn-efi; cd $TOPDIR/acrn-efi
+   $ echo > hv_cmdline.txt
+   $ echo RT_bzImage    > vm0_tag.txt
+   $ echo Linux_bzImage > vm1_tag.txt
+   $ echo ACPI_VM0      > acpi_vm0.txt
+
+   $ python3 GenContainer.py create -cl \
+     CMDL:./hv_cmdline.txt \
+     ACRN:$TOPDIR/acrn-hypervisor/build/hypervisor/acrn.32.out \
+     MOD0:./vm0_tag.txt  \
+     MOD1:./vm0_kernel   \
+     MOD2:./vm1_tag.txt  \
+     MOD3:./vm1_kernel   \
+     MOD4:./acpi_vm0.txt \
+     MOD5:$TOPDIR/acrn-hypervisor/build/hypervisor/acpi/ACPI_VM0.bin \
+     -o sbl_os    \
+     -t MULTIBOOT \
+     -a NONE
+
+You may optionally put HV boot options in the hv_cmdline.txt. Since the current implementation does not accept empty files,
+it must contain at least one character even if you don't need additional boot options.
+
+.. code-block:: none
+
+   # Acceptable Examples
+   $ echo     > hv_cmdline.txt    # end-of-line
+   $ echo " " > hv_cmdline.txt    # space + end-of-line
+
+   # Not Acceptable Example
+   $ touch hv_cmdline.txt         # empty file
```

Comment:
Added descriptions of hv_cmdline.txt since it turned out that it should not be an empty file.